plugins/woocommerce/includes/class-wc-discounts.php

```diff
@@ -742,7 +744,8 @@ protected function validate_coupon_maximum_amount( $coupon ) {
 	 * @return bool
 	 */
 	protected function validate_coupon_product_ids( $coupon ) {
-		if ( count( $coupon->get_product_ids() ) > 0 ) {
+		$product_ids_count = is_countable( $coupon->get_product_ids() ) ? count( $coupon->get_product_ids() ) : 0;
+		if ( $product_ids_count > 0 ) {
```

Comment:
Let's revert this change.  It's okay to assume the prop is an array getting returned.  In the case that it isn't it is probably best that a fatal stops execution - the same goes for the rest of the getter calls in this file.

Reply:
Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c